### ISSUE TITLE
[JENKINS-23083] Adding a second project to the list to trigger causes NPE

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#23118]
<p>Adding a second project to the projects to build in 'Post build Actions'<br/>
results in only the first project being triggered and this message concerning an NPE at the end of the parent log.</p>

<p>BUILD SUCCESSFUL<br/>
Total time: 7 minutes 40 seconds<br/>
Warning: you have no plugins providing access control for builds, so falling back to legacy behavior of permitting any downstream builds to be triggered<br/>
Triggering a new build of prowess_intel #651<br/>
FATAL: null<br/>
java.lang.NullPointerException<br/>
	at hudson.model.Result.isBetterOrEqualTo(Result.java:125)<br/>
	at jenkins.triggers.ReverseBuildTrigger.shouldTrigger(ReverseBuildTrigger.java:133)<br/>
	at jenkins.triggers.ReverseBuildTrigger.access$000(ReverseBuildTrigger.java:83)<br/>
	at jenkins.triggers.ReverseBuildTrigger$1.shouldTriggerBuild(ReverseBuildTrigger.java:140)<br/>
	at hudson.tasks.BuildTrigger.execute(BuildTrigger.java:243)<br/>
	at hudson.model.AbstractBuild$AbstractBuildExecution.cleanUp(AbstractBuild.java:674)<br/>
	at hudson.model.Build$BuildExecution.cleanUp(Build.java:193)<br/>
	at hudson.model.Run.execute(Run.java:1752)<br/>
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)<br/>
	at hudson.model.ResourceController.execute(ResourceController.java:88)<br/>
	at hudson.model.Executor.run(Executor.java:231)</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/chris_jones.jpg" title="chris_jones's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=chris_jones">chris_jones</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23083" target="_blank">Adding a second project to the list to trigger causes NPE</a></i></summary>
<i><ul>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: core
<li><b>resolution</b>: Duplicate
<li><b>resolved</b>: 2014-07-04T00:29:50+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 5
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Adding a second project to the projects to build in 'Post build Actions'
results in only the first project being triggered and this message concerning an NPE at the end of the parent log.</p>

<p>BUILD SUCCESSFUL
Total time: 7 minutes 40 seconds
Warning: you have no plugins providing access control for builds, so falling back to legacy behavior of permitting any downstream builds to be triggered
Triggering a new build of prowess_intel #651
FATAL: null
java.lang.NullPointerException
	at hudson.model.Result.isBetterOrEqualTo(Result.java:125)
	at jenkins.triggers.ReverseBuildTrigger.shouldTrigger(ReverseBuildTrigger.java:133)
	at jenkins.triggers.ReverseBuildTrigger.access$000(ReverseBuildTrigger.java:83)
	at jenkins.triggers.ReverseBuildTrigger$1.shouldTriggerBuild(ReverseBuildTrigger.java:140)
	at hudson.tasks.BuildTrigger.execute(BuildTrigger.java:243)
	at hudson.model.AbstractBuild$AbstractBuildExecution.cleanUp(AbstractBuild.java:674)
	at hudson.model.Build$BuildExecution.cleanUp(Build.java:193)
	at hudson.model.Run.execute(Run.java:1752)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:231)</p>
</pre>
</details>
</details>
<ul><li><i>environment</i>: <code>RHEL 6.5 X86_64 Jenkins installed from RPM and updated via yum to 1.563</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23083?focusedId=200639&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-200639">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>It looks more like the downstream project has a reverse trigger ("Trigger after other project completes" or so). Re-save both projects' configuration to get rid of any lingering brokenness due to the fix for <a href="https://issues.jenkins.io/browse/JENKINS-16956" title="Require authentication for build triggers" class="issue-link" data-issue-key="JENKINS-16956"><del>JENKINS-16956</del></a>.</p>
</pre>
</details>

</ul>
</details>

<p>It looks more like the downstream project has a reverse trigger ("Trigger after other project completes" or so). Re-save both projects' configuration to get rid of any lingering brokenness due to the fix for <a href="https://issues.jenkins.io/browse/JENKINS-16956" title="Require authentication for build triggers" class="issue-link" data-issue-key="JENKINS-16956"><del>JENKINS-16956</del></a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>spyder:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23083?focusedId=205100&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205100">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>OK, it wasn't that difficult after all.</p>

<p>We only recently upgraded from a build older than 1.560, so we are only just starting to use Jenkins with the build trigger change in <a href="https://issues.jenkins.io/browse/JENKINS-16956" title="Require authentication for build triggers" class="issue-link" data-issue-key="JENKINS-16956"><del>JENKINS-16956</del></a>.</p>

<p>The project causing the NPE had a bunch of projects in it's "build after other projects are built" trigger. This happened because we are still habitually using the previous behaviour where that list mirrored the post-build action "build other projects".</p>

<p>I re-saved the project causing the NPE, and all projects in the "build after other projects" trigger, but the NPE persisted. Only once I deleted the "other projects" trigger did things return to normal.</p>


<p>As far as I'm seeing, the reverse trigger is now broken and only the post-build action can be used. We're using 1.570.</p>
</pre>
</details>

</ul>
</details>

<p>OK, it wasn't that difficult after all.</p>

<p>We only recently upgraded from a build older than 1.560, so we are only just starting to use Jenkins with the build trigger change in <a href="https://issues.jenkins.io/browse/JENKINS-16956" title="Require authentication for build triggers" class="issue-link" data-issue-key="JENKINS-16956"><del>JENKINS-16956</del></a>.</p>

<p>The project causing the NPE had a bunch of projects in it's "build after other projects are built" trigger. This happened because we are still habitually using the previous behaviour where that list mirrored the post-build action "build other projects".</p>

<p>I re-saved the project causing the NPE, and all projects in the "build after other projects" trigger, but the NPE persisted. Only once I deleted the "other projects" trigger did things return to normal.</p>


<p>As far as I'm seeing, the reverse trigger is now broken and only the post-build action can be used. We're using 1.570.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>spyder:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23083?focusedId=205129&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205129">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Yes. Using the same context as <a href="https://issues.jenkins.io/browse/JENKINS-23191" title="ReverseBuildTrigger.threshold not consistently saved" class="issue-link" data-issue-key="JENKINS-23191"><del>JENKINS-23191</del></a> for clarity:</p>

<p>The reverse build trigger now saves correctly, but when Job A finishes, and attempts to trigger Job B, it gets the NPE described in this case (and, I believe, 23191).</p>

<p>We just added a new Job B type to our system, and had to go around to all of the Job A types it depends on and update the post-build trigger to include Job B.</p>
</pre>
</details>

</ul>
</details>

<p>Yes. Using the same context as <a href="https://issues.jenkins.io/browse/JENKINS-23191" title="ReverseBuildTrigger.threshold not consistently saved" class="issue-link" data-issue-key="JENKINS-23191"><del>JENKINS-23191</del></a> for clarity:</p>

<p>The reverse build trigger now saves correctly, but when Job A finishes, and attempts to trigger Job B, it gets the NPE described in this case (and, I believe, 23191).</p>

<p>We just added a new Job B type to our system, and had to go around to all of the Job A types it depends on and update the post-build trigger to include Job B.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23083?focusedId=205167&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205167">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This really sounds like a duplicate of <a href="https://issues.jenkins.io/browse/JENKINS-23191" title="ReverseBuildTrigger.threshold not consistently saved" class="issue-link" data-issue-key="JENKINS-23191"><del>JENKINS-23191</del></a>. Can you reproduce this <em>from scratch</em> in a new Jenkins installation containing that fix? Or is it just a matter of your having saved invalid configuration in the past due to <a href="https://issues.jenkins.io/browse/JENKINS-23191" title="ReverseBuildTrigger.threshold not consistently saved" class="issue-link" data-issue-key="JENKINS-23191"><del>JENKINS-23191</del></a> and have not since corrected the job configuration in all cases?</p>
</pre>
</details>

</ul>
</details>

<p>This really sounds like a duplicate of <a href="https://issues.jenkins.io/browse/JENKINS-23191" title="ReverseBuildTrigger.threshold not consistently saved" class="issue-link" data-issue-key="JENKINS-23191"><del>JENKINS-23191</del></a>. Can you reproduce this <em>from scratch</em> in a new Jenkins installation containing that fix? Or is it just a matter of your having saved invalid configuration in the past due to <a href="https://issues.jenkins.io/browse/JENKINS-23191" title="ReverseBuildTrigger.threshold not consistently saved" class="issue-link" data-issue-key="JENKINS-23191"><del>JENKINS-23191</del></a> and have not since corrected the job configuration in all cases?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>spyder:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23083?focusedId=205174&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205174">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Or, as the description of this case says, it could be having more than one project in the list.</p>

<p>I think I have a test Jenkins server floating around somewhere, I'll see what I can do. I'm under a bit of a tight deadline at the moment though.</p>
</pre>
</details>

</ul>
</details>

<p>Or, as the description of this case says, it could be having more than one project in the list.</p>

<p>I think I have a test Jenkins server floating around somewhere, I'll see what I can do. I'm under a bit of a tight deadline at the moment though.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23083?focusedId=205175&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205175">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Cannot reproduce on 1.570 with 3 freestyle projects ("up", "up2", "down") created from scratch:</p>

<ul class="alternate" type="square">
	<li>configure "down" to be triggered after "up, up2" are finished</li>
	<li>build "up2"</li>
</ul>


<p>build log of up2:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Started by user Daniel Beck
[EnvInject] - Loading node environment variables.
Building on master in workspace /<span class="code-keyword">var</span>/lib/jenkins/workspace/JENKINS-23083-up2
Warning: you have no plugins providing access control <span class="code-keyword">for</span> builds, so falling back to legacy behavior of permitting any downstream builds to be triggered
Triggering a <span class="code-keyword">new</span> build of JENKINS-23083-down
Finished: SUCCESS</pre>
</div></div>

<p>build log of down:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Started by upstream project <span class="code-quote">"JENKINS-23083-up2"</span> build number 1
originally caused by:
 Started by user Daniel Beck
[EnvInject] - Loading node environment variables.
Building on master in workspace /<span class="code-keyword">var</span>/lib/jenkins/workspace/JENKINS-23083-down
Finished: SUCCESS</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>Cannot reproduce on 1.570 with 3 freestyle projects ("up", "up2", "down") created from scratch:</p>

<ul class="alternate" type="square">
	<li>configure "down" to be triggered after "up, up2" are finished</li>
	<li>build "up2"</li>
</ul>


<p>build log of up2:</p>

<pre>
Started by user Daniel Beck
[EnvInject] - Loading node environment variables.
Building on master in workspace /<span class="code-keyword">var</span>/lib/jenkins/workspace/JENKINS-23083-up2
Warning: you have no plugins providing access control <span class="code-keyword">for</span> builds, so falling back to legacy behavior of permitting any downstream builds to be triggered
Triggering a <span class="code-keyword">new</span> build of JENKINS-23083-down
Finished: SUCCESS</pre>

<p>build log of down:</p>

<pre>
Started by upstream project <span class="code-quote">"JENKINS-23083-up2"</span> build number 1
originally caused by:
 Started by user Daniel Beck
[EnvInject] - Loading node environment variables.
Building on master in workspace /<span class="code-keyword">var</span>/lib/jenkins/workspace/JENKINS-23083-down
Finished: SUCCESS</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>spyder:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23083?focusedId=205176&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205176">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks for that <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>My understanding of <a href="https://issues.jenkins.io/browse/JENKINS-23191" title="ReverseBuildTrigger.threshold not consistently saved" class="issue-link" data-issue-key="JENKINS-23191"><del>JENKINS-23191</del></a> is that it required 'down' to have a post-build action as well as a 'build after' trigger. Maybe that is required here as well? It's certainly the case in my setup.</p>
</pre>
</details>

</ul>
</details>

<p>Thanks for that <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>My understanding of <a href="https://issues.jenkins.io/browse/JENKINS-23191" title="ReverseBuildTrigger.threshold not consistently saved" class="issue-link" data-issue-key="JENKINS-23191"><del>JENKINS-23191</del></a> is that it required 'down' to have a post-build action as well as a 'build after' trigger. Maybe that is required here as well? It's certainly the case in my setup.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23083?focusedId=205177&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205177">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Andrew: Still nothing. "up2" triggers "down" triggers "downer", all successful. </p>

<p>Make sure that in your "down" (or rather "center") job config, one of the radio buttons of the "build after other" trigger is selected. That could be a leftover from the original bug.</p>
</pre>
</details>

</ul>
</details>

<p>Andrew: Still nothing. "up2" triggers "down" triggers "downer", all successful. </p>

<p>Make sure that in your "down" (or rather "center") job config, one of the radio buttons of the "build after other" trigger is selected. That could be a leftover from the original bug.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>spyder:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23083?focusedId=205179&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205179">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Sadly, doesn't help. I might as well start talking real codenames to make it easier for me <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>The new project I added yesterday is called limbo. It has about 5 projects that it depends on, and one that it triggers. We manually added limbo to the post-build action of each parent project after the bug yesterday.</p>

<p>I had removed the "build after" trigger completely from limbo. I'm not sure if the radio button was selected or not. I just re-added it, confirmed the top radio button was selected and then triggered one of the parents in the list - a build called sugar. Sugar is one of our central projects, it has over 20 projects that depend on it (and are therefore listed in the post-build action).</p>

<p>From sugar's build log:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">09:32:30 Archiving artifacts
09:32:30 Recording test results
09:32:39 Notified Stash <span class="code-keyword">for</span> commit with id 329f2e748617ed52a00cf8be1512417714747f9d
09:32:39 Recording fingerprints
09:32:40 Warning: you have no plugins providing access control <span class="code-keyword">for</span> builds, so falling back to legacy behavior of permitting any downstream builds to be triggered
09:32:40 Triggering a <span class="code-keyword">new</span> build of polish
09:32:40 Triggering a <span class="code-keyword">new</span> build of gel
09:32:40 sugar-tunic is disabled. Triggering skipped
09:32:40 Triggering a <span class="code-keyword">new</span> build of tral
09:32:40 Triggering a <span class="code-keyword">new</span> build of blab
09:32:40 Triggering a <span class="code-keyword">new</span> build of cereal
09:32:40 Triggering a <span class="code-keyword">new</span> build of snooker
09:32:40 Triggering a <span class="code-keyword">new</span> build of dragster
09:32:40 FATAL: <span class="code-keyword">null</span>
09:32:40 java.lang.NullPointerException
09:32:40 	at hudson.model.Result.isBetterOrEqualTo(Result.java:125)
09:32:40 	at jenkins.triggers.ReverseBuildTrigger.shouldTrigger(ReverseBuildTrigger.java:133)
09:32:40 	at jenkins.triggers.ReverseBuildTrigger.access$000(ReverseBuildTrigger.java:83)
09:32:40 	at jenkins.triggers.ReverseBuildTrigger$1.shouldTriggerBuild(ReverseBuildTrigger.java:140)
09:32:40 	at hudson.tasks.BuildTrigger.execute(BuildTrigger.java:243)
09:32:40 	at hudson.model.AbstractBuild$AbstractBuildExecution.cleanUp(AbstractBuild.java:701)
09:32:40 	at hudson.model.Build$BuildExecution.cleanUp(Build.java:194)
09:32:40 	at hudson.model.Run.execute(Run.java:1778)
09:32:40 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
09:32:40 	at hudson.model.ResourceController.execute(ResourceController.java:88)
09:32:40 	at hudson.model.Executor.run(Executor.java:234)
</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>Sadly, doesn't help. I might as well start talking real codenames to make it easier for me <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>The new project I added yesterday is called limbo. It has about 5 projects that it depends on, and one that it triggers. We manually added limbo to the post-build action of each parent project after the bug yesterday.</p>

<p>I had removed the "build after" trigger completely from limbo. I'm not sure if the radio button was selected or not. I just re-added it, confirmed the top radio button was selected and then triggered one of the parents in the list - a build called sugar. Sugar is one of our central projects, it has over 20 projects that depend on it (and are therefore listed in the post-build action).</p>

<p>From sugar's build log:</p>


<pre>
09:32:30 Archiving artifacts
09:32:30 Recording test results
09:32:39 Notified Stash <span class="code-keyword">for</span> commit with id 329f2e748617ed52a00cf8be1512417714747f9d
09:32:39 Recording fingerprints
09:32:40 Warning: you have no plugins providing access control <span class="code-keyword">for</span> builds, so falling back to legacy behavior of permitting any downstream builds to be triggered
09:32:40 Triggering a <span class="code-keyword">new</span> build of polish
09:32:40 Triggering a <span class="code-keyword">new</span> build of gel
09:32:40 sugar-tunic is disabled. Triggering skipped
09:32:40 Triggering a <span class="code-keyword">new</span> build of tral
09:32:40 Triggering a <span class="code-keyword">new</span> build of blab
09:32:40 Triggering a <span class="code-keyword">new</span> build of cereal
09:32:40 Triggering a <span class="code-keyword">new</span> build of snooker
09:32:40 Triggering a <span class="code-keyword">new</span> build of dragster
09:32:40 FATAL: <span class="code-keyword">null</span>
09:32:40 java.lang.NullPointerException
09:32:40 	at hudson.model.Result.isBetterOrEqualTo(Result.java:125)
09:32:40 	at jenkins.triggers.ReverseBuildTrigger.shouldTrigger(ReverseBuildTrigger.java:133)
09:32:40 	at jenkins.triggers.ReverseBuildTrigger.access$000(ReverseBuildTrigger.java:83)
09:32:40 	at jenkins.triggers.ReverseBuildTrigger$1.shouldTriggerBuild(ReverseBuildTrigger.java:140)
09:32:40 	at hudson.tasks.BuildTrigger.execute(BuildTrigger.java:243)
09:32:40 	at hudson.model.AbstractBuild$AbstractBuildExecution.cleanUp(AbstractBuild.java:701)
09:32:40 	at hudson.model.Build$BuildExecution.cleanUp(Build.java:194)
09:32:40 	at hudson.model.Run.execute(Run.java:1778)
09:32:40 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
09:32:40 	at hudson.model.ResourceController.execute(ResourceController.java:88)
09:32:40 	at hudson.model.Executor.run(Executor.java:234)
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>spyder:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23083?focusedId=205180&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205180">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>My apologies, it wasn't the project we created yesterday. We'd been running the buggy build of Jenkins long enough that two of the other projects which depend on Sugar had their "build after" trigger updated recently.</p>

<p>Saving those fixed the issue.</p>

<p>There are two things that make this a nightmare to track down:</p>
<ul>
	<li>When the NPE occurs, there is no indication in the log of which project caused it</li>
	<li>The build that experiences the NPE is marked as successful so we never see a notification that there is an issue</li>
</ul>


<p>If we had that, this issue would be much easier to recover from.</p>
</pre>
</details>

</ul>
</details>

<p>My apologies, it wasn't the project we created yesterday. We'd been running the buggy build of Jenkins long enough that two of the other projects which depend on Sugar had their "build after" trigger updated recently.</p>

<p>Saving those fixed the issue.</p>

<p>There are two things that make this a nightmare to track down:</p>
<ul>
	<li>When the NPE occurs, there is no indication in the log of which project caused it</li>
	<li>The build that experiences the NPE is marked as successful so we never see a notification that there is an issue</li>
</ul>


<p>If we had that, this issue would be much easier to recover from.</p>
